From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: violetc <58360096+s-yh-china@users.noreply.github.com>
Date: Fri, 7 Feb 2025 22:32:29 +0800
Subject: [PATCH] Dont save Entity tickCount

Storing just tickCount will only lead to incorrect entity behavior and will not bring any benefits.

diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index caddf2c1dcaca7884dbdb6af8311c6bfb9a7aa60..14b09ee5d605ec8ff86165fdc0b7aeba310f833e 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -2464,7 +2464,6 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
             if (this.maxAirTicks != this.getDefaultMaxAirSupply()) {
                 compound.putInt("Bukkit.MaxAirSupply", this.getMaxAirSupply());
             }
-            compound.putInt("Spigot.ticksLived", this.tickCount);
             // CraftBukkit end
             Component customName = this.getCustomName();
             if (customName != null) {
@@ -2637,11 +2636,6 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                 throw new IllegalStateException("Entity has invalid rotation");
             }
             // CraftBukkit start
-            // Spigot start
-            if (this instanceof net.minecraft.world.entity.LivingEntity) {
-                this.tickCount = compound.getInt("Spigot.ticksLived");
-            }
-            // Spigot end
             this.persist = !compound.contains("Bukkit.persist") || compound.getBoolean("Bukkit.persist");
             this.visibleByDefault = !compound.contains("Bukkit.visibleByDefault") || compound.getBoolean("Bukkit.visibleByDefault");
             // SPIGOT-6907: re-implement LivingEntity#setMaximumAir()
